### PR TITLE
Add core http client

### PR DESCRIPTION
**Pull request by abantecart**

### CODE REVIEW

File: abantecart/abc/core/lib/GuzzleHttpClient.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?php
```
Comment (abolabo) [inline abantecart/abc/core/lib/GuzzleHttpClient.php:1]: why this file here? i think it must be placed inside abc/vendor directory and loads via autoloader of composer

File: abantecart/abc/core/lib/HttpClient.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (abolabo) [inline abantecart/abc/core/lib/HttpClient.php:1]: the same question

File: abantecart/abc/core/lib/HttpResponse.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?php
```
Comment (abolabo) [inline abantecart/abc/core/lib/HttpResponse.php:1]: the same question

**Review by abolabo — CHANGES_REQUESTED** [review]

File: abantecart/abc/core/lib/GuzzleHttpClient.php (hunk 1 above)
Comment (vdvpie22) [reply abantecart/abc/core/lib/GuzzleHttpClient.php:1]: This is implementation of interface. That used Guzzle, Guzzle stored in vendor directory.

File: abantecart/abc/core/lib/HttpClient.php (hunk 2 above)
Comment (vdvpie22) [reply abantecart/abc/core/lib/HttpClient.php:1]: We add core requirements for http client, we can use different implementations based on: curl, guzzle etc.

File: abantecart/abc/core/lib/HttpResponse.php (hunk 3 above)
Comment (vdvpie22) [reply abantecart/abc/core/lib/HttpResponse.php:1]: Same answer

**Pull request merged**